ui/web/static/js/charts.js

```diff
@@ -1658,23 +1867,93 @@ function createTempGaugeChart(chartData, config) {
       }, 2000);
 
     tempGaugeChart.setOption(option);`;
-  return config;
+  }
 }
 
-function createValueCard(chartData, config) {
-  config.Content = `
+class ValueCard {
+  constructor(chartData, widgetID) {
+    this.chartData = chartData;
+    this.ID = widgetID;
+    this.Style = {
+      width: "400px",
+      height: "200px",
+    };
+    let styleString = `width: ${this.Style.width}; height: ${this.Style.height};`;
+    this.Content = `
+    <div class="item-content" id="${this.ID}" style="${styleString}">
     <div class="card mt-3 widgetcard" >
       <div class="card-header text-center">
-        <h5 class="card-title">${chartData.title}</h5>
+        <h5 class="card-title">${this.chartData.title}</h5>
       </div>
       <div class="card-body text-center">
         <p class="card-text value"> 35</p>
       </div>
       <div class="card-footer">
-        <p class="card-text"> Units: ${chartData.valueUnits}</p>
-        <p class="card-text text"> ThingName: ${chartData.thingName}</p>
+        <p class="card-text"> Units: ${this.chartData.valueUnits}</p>
+        <p class="card-text text"> ThingName: ${this.chartData.thingName}</p>
       </div>
     </div>
+    </div>
   `;
-  return config;
+  }
+}
+
+const chartTypes = {
+  alarmCount: AlarmCount,
+  alarmsTable: AlarmsTable,
+  areaLineChart: AreaLineChart,
+  timeSeriesBarChart: TimeSeriesBarChart,
+  donutChart: DonutChart,
+  doubleBarChart: DoubleBarChart,
+  dynamicDataChart: DynamicDataChart,
+  entitiesTable: EntitiesTable,
+  entityCount: EntityCount,
+  gaugeChart: GaugeChart,
+  horizontalBarChart: HorizontalBarChart,
+  labelCard: LabelCard,
+  lineChart: TimeSeriesLineChart,
+  multiBarChart: MultiBarChart,
+  multiGaugeChart: MultiGaugeChart,
+  multipleLineChart: MultipleLineChart,
+  pieChart: PieChart,
+  progressBar: ProgressBar,
+  sharedDatasetChart: SharedDatasetChart,
+  speedGaugeChart: SpeedGaugeChart,
+  stackedLineChart: StackedLineChart,
+  stepChart: StepChart,
+  tempGaugeChart: TempGaugeChart,
+  valueCard: ValueCard,
+};
+
+class Widget {
+  constructor(chartData, widgetID) {
+    this.chartData = chartData;
+    this.widgetID = widgetID;
+    this.config = new chartTypes[this.chartData.Type](chartData, widgetID);
```

Comment:
I don't think there is any need for the chartTypes above. You can pass the actual chart in the chartData.Type and it will be sth like

```suggestion
    this.config = new this.chartData.Type(chartData, widgetID);
```

Reply:
The charttype doesnt match the class name.

Reply:
You can change in the individual charts so they match e.g instead of `chartData["Type"] = "alarmCount";` you have `chartData["Type"] = "AlarmCount";`